pkg/sdk/file_format.go

```diff
@@ -0,0 +1,1038 @@
+	AvroCompressionBrotli     AvroCompression = "BROTLI"
+	AvroCompressionZstd       AvroCompression = "ZSTD"
+	AvroCompressionDeflate    AvroCompression = "DEFLATE"
+	AvroCompressionRawDeflate AvroCompression = "RAW_DEFLATE"
+	AvroCompressionNone       AvroCompression = "NONE"
+)
+
+type ParquetCompression string
+
+var (
+	ParquetCompressionAuto   ParquetCompression = "AUTO"
+	ParquetCompressionLzo    ParquetCompression = "LZO"
+	ParquetCompressionSnappy ParquetCompression = "SNAPPY"
+	ParquetCompressionNone   ParquetCompression = "NONE"
+)
+
+type XmlCompression string
+
+var (
+	XmlCompressionAuto       XmlCompression = "AUTO"
+	XmlCompressionGzip       XmlCompression = "GZIP"
+	XmlCompressionBz2        XmlCompression = "BZ2"
+	XmlCompressionBrotli     XmlCompression = "BROTLI"
+	XmlCompressionZstd       XmlCompression = "ZSTD"
+	XmlCompressionDeflate    XmlCompression = "DEFLATE"
+	XmlCompressionRawDeflate XmlCompression = "RAW_DEFLATE"
+	XmlCompressionNone       XmlCompression = "NONE"
+)
+
+type NullString struct {
+	S string `ddl:"parameter,no_equals,single_quotes"`
+}
+
+type CreateFileFormatOptions struct {
+	create      bool                   `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace   *bool                  `ddl:"keyword" sql:"OR REPLACE"`
+	Temporary   *bool                  `ddl:"keyword" sql:"TEMPORARY"`
+	fileFormat  bool                   `ddl:"static" sql:"FILE FORMAT"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists *bool                  `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name        SchemaObjectIdentifier `ddl:"identifier"`
+	Type        FileFormatType         `ddl:"parameter" sql:"TYPE"`
+
+	FileFormatTypeOptions
+}
+
+func (opts *CreateFileFormatOptions) validate() error {
+	fields := opts.FileFormatTypeOptions.fieldsByType()
+
+	for formatType := range fields {
+		if opts.Type == formatType {
+			continue
+		}
+		if anyValueSet(fields[formatType]...) {
+			return fmt.Errorf("Cannot set %s fields when TYPE = %s", formatType, opts.Type)
+		}
+	}
+
+	err := opts.FileFormatTypeOptions.validate()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *fileFormats) Create(ctx context.Context, id SchemaObjectIdentifier, opts *CreateFileFormatOptions) error {
+	if opts == nil {
+		opts = &CreateFileFormatOptions{}
+	}
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type AlterFileFormatOptions struct {
+	alter      bool                   `ddl:"static" sql:"ALTER"`       //lint:ignore U1000 This is used in the ddl tag
+	fileFormat bool                   `ddl:"static" sql:"FILE FORMAT"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists   *bool                  `ddl:"keyword" sql:"IF EXISTS"`
+	name       SchemaObjectIdentifier `ddl:"identifier"`
+
+	Rename *AlterFileFormatRenameOptions
```

Comment:
I found examples of both names in the code base, I think I've been using Rename and you've been using NewName. Happy to go with either, I went with Rename because it's closed to the sql.